[4.6.0] Add docs for external WebSub hub and custom WebSub topic matcher (#11571)
wso2/docs-apim: en/docs/install-and-setup/setup/advance-configurations/configuring-external-websub-hub.md

# Configuring an External WebSub Hub

> By default, WSO2 API Manager handles the WebSub protocol (topic registration, subscription, unsubscription, and content publishing) inside the Universal Gateway itself. This works well for most deployments, but there are situations where you may want to delegate the hub responsibility to a dedicated external WebSub hub — for example, to reuse an existing hub that already serves other systems, to scale hub traffic independently of API traffic, or to take advantage of hub-specific features such as persistent subscriptions.

Reviewer: ```suggestion
By default, WSO2 API Manager handles the WebSub protocol (topic registration, subscription, unsubscription, and content publishing) inside the Universal Gateway itself. This works well for most deployments, but there are situations where you may want to delegate the hub responsibility to a dedicated external WebSub hub — for example, to reuse an existing hub that already serves other systems, to scale hub traffic independently of API traffic, or to take advantage of hub-specific features such as guaranteed deliveries.
```

> When an external hub is configured, WebSub APIs deployed from the Publisher continue to handle authentication, authorization, and throttling at the Gateway as usual, but the hub-mode requests (`subscribe`, `unsubscribe`, and `publish`) are forwarded to the external hub instead of being processed internally.

Reviewer: ```suggestion
When an external hub is configured, WebSub APIs deployed from the Publisher continue to handle authentication, authorization, and throttling at the Gateway as usual, but the hub requests (`subscribe`, `unsubscribe`, and `publish`) are forwarded to the external hub instead of being processed internally.
```

<th><b>Description</b></th>
<th><b>Required</b></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><p><code>url</code></p></td>
<td><p>The base HTTP(S) URL of your external WebSub hub. Setting this value activates external-hub delegation for newly deployed WebSub APIs. If the URL is empty or the section is omitted, the gateway uses its built-in hub.</p></td>
<td><p>Yes</p></td>
</tr>
<tr class="even">
<td><p><code>type</code></p></td>
<td><p>An identifier for the hub implementation (for example, <code>external</code>). Informational; used by custom extensions if you need to branch on hub type.</p></td>
<td><p>No</p></td>
</tr>
<tr class="odd">
<td><p><code>disable_topic_context_prefix</code></p></td>
<td><p>Controls how topic names are sent to the external hub.
<br/><br/>
<code>false</code> (default) — topics are prefixed with the API context to avoid collisions between different APIs that expose the same topic name.
<br/><br/>
<code>true</code> — topics are sent to the hub verbatim. Use this when the external hub already provides namespacing, or when multiple APIs must intentionally share the same topic.</p></td>
<td><p>No</p></td>
</tr>
</tbody>
</table>

### Topic naming

> When the topic-context prefix is enabled (the default), the gateway rewrites each topic before sending it to the external hub, so that topics from different APIs do not collide on a shared hub. The rewritten topic uses the API context as a prefix, with `/` characters replaced by `_`.

Reviewer: ```suggestion
When the topic-context prefix is enabled (the default), the gateway rewrites each topic before sending it to the external hub, so that topics with the same name from different WebSub APIs do not collide on a shared hub. The rewritten topic uses the API context as a prefix, with `/` characters replaced by `_`.
```

> For example, an API deployed at `/repo-watcher/1.0.0` publishing to topic `commits` is forwarded to the external hub as `repo-watcher_1.0.0_commits`. Publish and subscribe requests are rewritten the same way, so publishers and subscribers stay consistent.

Reviewer: ```suggestion
For example, the publish request to the `commits` topic of the `/repo-watcher/1.0.0` WebSub API is forwarded to the `repo-watcher_1.0.0_commits` topic of the external WebSub Hub. Publish and subscribe requests are rewritten the same way, so publishers and subscribers stay consistent.
```
